test/e2e/util_test.go

```diff
@@ -520,6 +520,31 @@ func probe(timeout, period, success, failure int) *corev1.Probe {
 	}
 }
 
+// updateRouteWithRetryOnConflict gets a fresh copy of the named route,
+// calls mutateRouteFn() where callers can modify fields of the route,
+// and then updates the route object. If there is a conflict error
+// on update then the complete sequence of get, mutate, and update
+// is retried until timeout is reached.
+func updateRouteWithRetryOnConflict(t *testing.T, name types.NamespacedName, timeout time.Duration, mutateRouteFn func(route *routev1.Route)) error {
+	t.Helper()
+	route := routev1.Route{}
```

Comment:
```suggestion
	route := &routev1.Route{}
```
Then there will be no need to take its address 3 times down the function.

Reply:
I suppose it's a matter of preference. Personally, I like to avoid using &. Why? Because when you have to take the address multiple times within a function, it's a useful signal to me that the route might be, or could be, mutated—even if it doesn't actually get mutated. Maybe that's just the C programmer in me...

Reply:
Thank you. I opted to keep taking the address in each line there. Keeping it in line with other helpers in test/e2e/util_test.go I suppose. I hope that's alright.

Reply:
I think that taking address once at initialization is more go idiomatic. In the code of this operator, we can this usage is more popular:
```
cluster-ingress-operator$ agi ' &routev1.Route{' -c | cut -d ':' -f2 | sum
26

cluster-ingress-operator$ agi ' routev1.Route{' -c | cut -d ':' -f2 | sum
6

$ which agi
alias agi='ag --ignore-dir=vendor --ignore=tags --ignore=go.sum'
	/usr/bin/ag
```

as well as in some other highly used go projects like kube's client-go, for instance (taken from `vendor` of this repository too):
```
client-go$ ag ' &v1.Pod{' -c | cut -d ':' -f2 | sum
8
client-go$ ag ' &v1.Service{' -c | cut -d ':' -f2 | sum
7

client-go$ ag ' v1.Service{' || echo "not found"
not found
client-go$ ag ' v1.Pod{' || echo "not found"
not found
```

I agree that the pointer can signal a potential modification. However, unfortunately, in golang we never know for sure whether the intention was to save some copying or to modify the object. So the caller have to make sure anyway.

Reply:
Thank you for different perspectives on this. After further consideration I put this change back in as @alebedev87 suggested.